src/pages/version/taskDuration/TaskDurationTable.tsx

```diff
@@ -0,0 +1,279 @@
+                  ];
+                return (
+                  <Row key={task.id}>
+                    <TaskNameCell>
+                      <TaskLink taskId={task.id} taskName={task.displayName} />
+                      <TaskStatusIcon status={task.status} />
+                    </TaskNameCell>
+                    <BuildVariantCell>
+                      {task.buildVariantDisplayName}
+                    </BuildVariantCell>
+                    <DurationCell>
+                      <DurationBar width={barWidth} color={barColor} />
+                      <DurationLabel>
+                        {msToDuration(task.timeTaken)}
+                      </DurationLabel>
+                    </DurationCell>
+                  </Row>
+                );
+              })}
+          </DisplayTaskRow>
+        )}
+      </Table>
+      {tasks.length === 0 && <NoTableResults message="No tasks found." />}
+    </TableWrapper>
+  );
+};
+
+interface RowProps {
+  task: PatchTaskDurationsQuery["patchTasks"]["tasks"][0];
+  max: number;
+  "data-cy"?: string;
+  children?: React.ReactNode;
+}
+
+const DisplayTaskRow: React.VFC<RowProps> = ({
+  task,
+  max,
+  children,
+  "data-cy": dataCy,
+}) => {
+  const {
+    id,
+    displayName,
+    status,
+    buildVariantDisplayName,
+    timeTaken,
+    startTime,
+  } = task;
+
+  const barWidth = calculateBarWidth(timeTaken, max);
+  const barColor =
+    mapTaskToBarchartColor[mapTaskStatusToUmbrellaStatus[status]];
+
+  const startedWithZeroTime =
+    startTime === null && status === TaskStatus.Started;
+
+  return (
+    <Row key={id} data-cy={dataCy}>
+      <TaskNameCell>
+        <TaskLink taskId={id} taskName={displayName} />
+        <TaskStatusIcon status={status} />
+      </TaskNameCell>
+      <BuildVariantCell>{buildVariantDisplayName}</BuildVariantCell>
+      <DurationCell>
+        {startedWithZeroTime ? (
+          <DurationLabel>
+            There is no task duration information for this task at this time.
+          </DurationLabel>
+        ) : (
+          <>
+            <DurationBar width={barWidth} color={barColor} />
+            <DurationLabel>{msToDuration(timeTaken)}</DurationLabel>
+          </>
+        )}
+      </DurationCell>
+      {children}
+    </Row>
+  );
+};
+
+const findMax = (tasks: PatchTaskDurationsQuery["patchTasks"]["tasks"]) => {
+  const durations = tasks.map((t) => (t.startTime !== null ? t.timeTaken : 0));
+  return Math.max(...durations);
+};
+
+const calculateBarWidth = (value: number, max: number) =>
+  max ? `${(value / max) * 100}%` : "0%";
+
+const TableWrapper = styled.div`
+  border-top: 3px solid ${gray.light2};
+`;
+
+const StyledTableHeader = styled(TableHeader)`
+  width: 20%;
+`;
+
+const TableHeaderLabel = styled.div`
+  display: flex;
+  align-items: center;
+`;
+
+const TaskNameCell = styled(Cell)`
+  word-break: break-all;
+  padding-right: ${size.m};
+  svg {
+    margin-left: ${size.xxs};
+    vertical-align: text-bottom;
+  }
```

Comment:
You can apply a wrapper to the contents in the task Cell and do an `align-items: center` to align the task icons. (Which is what i assume this is doing)

Regardless i am not really a big fan of how this looks for larger task names. Can we just add another column for task status?